src/modules/DistributionManager.sol

```diff
@@ -0,0 +1,535 @@
+    uint256 public yieldFixedSplitDivisor;
+    uint256 public lastDistributionBlock;
+    uint256 public cycleNumber;
+    uint256 public totalYieldCollected;
+    uint256 public lastCollectionBlock;
+
+    address[] public recipients;
+    uint256[] public currentVotes;
+    uint256 public totalVotes;
+
+    mapping(uint256 => DistributionState) public distributionHistory;
+
+    event YieldCollected(address indexed source, uint256 amount, uint256 blockNumber);
+    event YieldValidated(uint256 totalYield);
+
+    modifier onlyEmergencyAdmin() {
+        if (msg.sender != emergencyAdmin && msg.sender != owner()) {
+            revert OnlyEmergencyAdmin();
+        }
+        _;
+    }
+
+    /// @notice Initializes the distribution manager
+    /// @param _yieldToken Address of the yield token
+    /// @param _yieldSource Address of the yield source
+    /// @param _cycleLength Initial cycle length in blocks
+    /// @param _yieldFixedSplitDivisor Initial fixed split divisor
+    function __DistributionManager_init(
+        address _yieldToken,
+        address _yieldSource,
+        uint256 _cycleLength,
+        uint256 _yieldFixedSplitDivisor
+    ) internal {
+        if (_yieldToken == address(0)) revert ZeroAddress();
+        if (_yieldSource == address(0)) revert ZeroAddress();
+        if (_cycleLength == 0) revert InvalidCycleLength();
+        if (_yieldFixedSplitDivisor == 0) revert InvalidDivisor();
+
+        yieldToken = _yieldToken;
+        yieldSource = _yieldSource;
+        cycleLength = _cycleLength;
+        yieldFixedSplitDivisor = _yieldFixedSplitDivisor;
+        lastDistributionBlock = block.number;
+
+        _initializeOwner(msg.sender);
+    }
+
+    /// @inheritdoc IDistributionModule
+    function distributeYield() external override nonReentrant whenNotPaused {
+        (bool canDistribute,) = validateDistribution();
+        if (!canDistribute) revert DistributionNotResolved();
+
+        // Hook for token minting before distribution
+        _mintTokensBeforeDistribution();
+
+        // Collect yield
+        uint256 totalYield = _collectYield();
+        if (totalYield == 0) revert InsufficientYield();
+
+        // Calculate splits
+        (uint256 fixedAmount, uint256 votedAmount) = _calculateSplits(totalYield);
+
+        // Get distribution data
+        address[] memory activeRecipients = _getActiveRecipients();
+        if (activeRecipients.length == 0) revert NoRecipients();
+
+        uint256[] memory votes = _getVotingResults();
+
+        // Calculate distributions
+        uint256[] memory votedDistributions = _calculateVotedDistributions(votes, totalVotes, votedAmount);
+        uint256[] memory fixedDistributions = _calculateFixedDistributions(activeRecipients, fixedAmount);
+
+        // Execute distributions
+        _executeDistributions(activeRecipients, votedDistributions, fixedDistributions);
+
+        // Complete cycle
+        _completeCycleTransition();
```

Comment:
should be removed from distribution manager, this is handelled by the cycle manager, add a mock and call it